detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/GoModCliExtractor.java

```diff
@@ -69,6 +77,22 @@ public Extraction extract(File directory, File goExe) {
         }
     }
 
+    private List<String> goListUJsonOutput(File directory, File goExe) throws ExecutableRunnerException, DetectableException {
+        List<String> versionOutput = execute(directory, goExe, "Querying for the version failed: ", "version");
+        Pattern pattern = Pattern.compile("\\d+\\.[\\d.]+");
```

Comment:
Also I believe Pattern.compile() is expensive and should be done once.

Reply:
@JakeMathews it's currently only done once, no?